src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -2411,11 +2385,14 @@ void MacroAssembler::membar(uint32_t order_constraint) {
 // actually be used: you must use the Address that is returned. It
 // is up to you to ensure that the shift provided matches the size
 // of your data.
-Address MacroAssembler::form_address(Register Rd, Register base, long byte_offset) {
+Address MacroAssembler::form_address(Register Rd, Register base, int64_t byte_offset) {
   if (is_offset_in_range(byte_offset, 12)) { // 12: imm in range 2^12
     return Address(base, byte_offset);
   }
 
+  assert(Rd != noreg, "expecting a register");
+  guarantee(Rd != base, "should be different registers");
```

Comment:
Why not `assert`?

Reply:
Fixed. These two lines come from the original MacroAssembler::baseOffset32. I think assert_different_registers will also do here.